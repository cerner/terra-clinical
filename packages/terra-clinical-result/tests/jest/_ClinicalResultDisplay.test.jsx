import React from 'react';
import ClinicalResultDisplay from '../../src/_ClinicalResultDisplay';
import DefaultResult from '../../src/terra-dev-site/test/clinical-result/TestResults';

describe('ClinicalResultDisplay', () => {
  it('should render a default ClinicalResultDisplay', () => {
    const result = shallow(<ClinicalResultDisplay {...DefaultResult} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResultDisplay', () => {
    const result = shallow(<ClinicalResultDisplay {...DefaultResult} isTruncated />);
    expect(result).toMatchSnapshot();
  });

  it('should render a truncated ClinicalResultDisplay with icons moved', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} isTruncated isModified />);
    expect(result).toMatchSnapshot();
  });

  it('should pass certain props to Observation ', () => {
    const resultData = {
      ...DefaultResult,
      interpretation: 'critical',
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} hideUnit />);
    expect(result).toMatchSnapshot();
  });

  it('should render a modified result ', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} isModified />);
    expect(result).toMatchSnapshot();
  });

  it('should render a commented result ', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} hasComment />);
    expect(result).toMatchSnapshot();
  });

  it('should render an unverified result ', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} isUnverified />);
    expect(result).toMatchSnapshot();
  });

  it('should render a result with unit hidden', () => {
    const resultData = {
      ...DefaultResult,
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} hideUnit />);
    expect(result).toMatchSnapshot();
  });

  it('should render a result with a concept display', () => {
    const resultData = {
      ...DefaultResult,
      conceptDisplay: 'Temperature Oral',
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a result with a date time display', () => {
    const resultData = {
      ...DefaultResult,
      datetimeDisplay: 'Nov 23, 2019 13:31:31',
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} />);
    expect(result).toMatchSnapshot();
  });

  it('should render a entered-in-error status result', () => {
    const resultData = {
      ...DefaultResult,
      status: 'entered-in-error',
      interpretation: 'critical',
    };
    const result = shallow(<ClinicalResultDisplay {...resultData} />);
    expect(result).toMatchSnapshot();
  });
});
