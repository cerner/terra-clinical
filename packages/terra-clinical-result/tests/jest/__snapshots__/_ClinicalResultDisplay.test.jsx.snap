// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClinicalResultDisplay should pass certain props to Observation  1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          hideUnit={true}
          interpretation="critical"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons />
    </div>
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a commented result  1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons
        hasComment={true}
      />
    </div>
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a default ClinicalResultDisplay 1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons />
    </div>
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a entered-in-error status result 1`] = `
<Fragment>
  <div
    className="decorated-result-display status-in-error"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={true}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          interpretation={null}
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons />
    </div>
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a modified result  1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons
        isModified={true}
      />
    </div>
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a result with a concept display 1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons />
    </div>
  </div>
  <SecondaryDisplays
    conceptDisplay="Temperature Oral"
  />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a result with a date time display 1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons />
    </div>
  </div>
  <SecondaryDisplays
    datetimeDisplay="Nov 23, 2019 13:31:31"
  />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a result with unit hidden 1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          hideUnit={true}
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons />
    </div>
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a truncated ClinicalResultDisplay 1`] = `
<Fragment>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
    </div>
    <Icons />
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render a truncated ClinicalResultDisplay with icons moved 1`] = `
<Fragment>
  <div
    className="decorated-result-display truncated"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
    </div>
    <Icons
      isModified={true}
    />
  </div>
  <SecondaryDisplays />
</Fragment>
`;

exports[`ClinicalResultDisplay should render an unverified result  1`] = `
<Fragment>
  <div
    className="decorated-result-display"
  >
    <div
      className="result-display"
    >
      <ConditionalWrapper
        condition={false}
        key="del-Observation-111"
        wrapper={[Function]}
      >
        <Observation
          eventId="111"
          isUnverified={true}
          key="Observation-111"
          result={
            Object {
              "unit": "mL",
              "value": "12345.678",
            }
          }
        />
      </ConditionalWrapper>
      <Icons
        isUnverified={true}
      />
    </div>
  </div>
  <SecondaryDisplays />
</Fragment>
`;
